gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallableFactory.java

```diff
@@ -53,14 +53,19 @@
 /** Class with utility methods to create grpc-based direct callables. */
 @BetaApi("The surface for use by generated code is not stable yet and may change in the future.")
 public class GrpcCallableFactory {
-  private GrpcCallableFactory() {}
+  protected GrpcCallableFactory() {}
 
-  private static <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> createBaseUnaryCallable(
+  protected <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> createDirectUnaryCallable(
```

Comment:
I'd strongly prefer to find a compositional solution to this design problem over an inheritance-based solution

Reply:
What do you think about a more agnostic solution? I have updated to add a CallableFactory interface. Because the implementation of CallableFactory will be handwritten, it is up to the implementer whether they want to implement everything themselves, or make use of the existing GrpcCallableFactory via composition or inheritance. In this case, because very little is changed, I personally would opt for inheritance. WDYT?